search-service/src/test/kotlin/com/egm/stellio/search/service/EntityOperationServiceTests.kt

```diff
@@ -297,21 +331,30 @@ class EntityOperationServiceTests {
 
         coVerify { temporalEntityAttributeService.deleteTemporalAttributesOfEntity(firstEntityURI) }
         coVerify { temporalEntityAttributeService.deleteTemporalAttributesOfEntity(secondEntityURI) }
-
         coVerify {
             entityPayloadService.appendAttributes(eq(firstEntityURI), any(), false, sub)
         }
         coVerify {
             entityPayloadService.appendAttributes(eq(secondEntityURI), any(), false, sub)
         }
+        coVerify(exactly = 2) {
+            entityEventService.publishEntityReplaceEvent(any(), any(), any())
+        }
     }
 
     @Test
     fun `batch delete should return the list of deleted entity ids when deletion is successful`() = runTest {
         coEvery { entityPayloadService.deleteEntity(any()) } returns Unit.right()
         coEvery { authorizationService.removeRightsOnEntity(any()) } returns Unit.right()
+        coEvery { entityEventService.publishEntityDeleteEvent(any(), any()) } returns Job()
 
-        val batchOperationResult = entityOperationService.delete(setOf(firstEntityURI, secondEntityURI))
+        val batchOperationResult = entityOperationService.delete(
+            setOf(
+                mockkClass(EntityPayload::class) { every { entityId } returns firstEntityURI },
+                mockkClass(EntityPayload::class) { every { entityId } returns secondEntityURI },
+            ),
```

Comment:
here : `every { entityId } returns firstEntityURI` and `every { entityId } returns secondEntityURI` is because in `entityOperationService.delete()` entities Ids are extracted to delete each entity ?

Reply:
`entityOperationService.delete()` now receives a set of `EntityPayload` objects instead of entities ids and it reads the id of each entity so it needs to have some.